extensions/publisher-cves/src/stores/vulns.ts

```diff
@@ -0,0 +1,73 @@
+import { defineStore } from "pinia";
+import { ref } from "vue";
+
+export interface Vulnerability {
+  id: string;
+  versions: Record<string, any>;
+  ranges: Array<{
+    type: string;
+    events: Array<{
+      introduced?: string;
+      fixed?: string;
+    }>;
+  }>;
+  summary: string;
+  details: string;
+  modified: string;
+  published: string;
+}
+
+export interface VulnerabilityMap {
+  [packageName: string]: Vulnerability[];
+}
+
+export const useVulnsStore = defineStore("vulns", () => {
+  // State
+  const pypi = ref<VulnerabilityMap>({});
+  const cran = ref<VulnerabilityMap>({});
+  const isLoading = ref(false);
+  const error = ref<Error | null>(null);
+
+  // Track fetch status
+  const isFetched = ref(false);
```

Comment:
A lot of the stores have this `isFetched`, or similar, variable to avoid calling APIs multiple times. Each is only called once right now.